src/plugins/kibana_react/public/page_template/no_data_page/no_data_config/no_data_config_page.tsx

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import { EuiPageTemplate } from '@elastic/eui';
+import React from 'react';
+import classNames from 'classnames';
+import { NO_DATA_PAGE_TEMPLATE_PROPS, NoDataPage } from '../no_data_page';
+import { withSolutionNavbar } from '../../with_solution_navbar';
+import { KibanaPageTemplateProps } from '../../page_template';
+
+export const NoDataConfigPage = (props: KibanaPageTemplateProps) => {
+  const { className, noDataConfig } = props;
+
+  if (!noDataConfig) {
+    return null;
+  }
+
+  const template = NO_DATA_PAGE_TEMPLATE_PROPS.template;
+
+  const classes = classNames(
+    'kbnPageTemplate',
+    { [`kbnPageTemplate--${template}`]: template },
+    className
+  );
+
+  return (
+    <EuiPageTemplate
+      data-test-subj={props['data-test-subj']}
+      template={template}
+      className={classes}
+      {...NO_DATA_PAGE_TEMPLATE_PROPS}
+    >
+      <NoDataPage {...noDataConfig} />
+    </EuiPageTemplate>
+  );
+};
+
+export const NoDataConfigPageWithSolutionNavBar = withSolutionNavbar(NoDataConfigPage);
```

Comment:
To be honest this whole pattern of wrapping components has got me confused because it causes a lot of obfuscation. But I'll leave that to @clintandrewhall 's discretion for review.

Reply:
Higher order components are a common pattern in [React](https://reactjs.org/docs/higher-order-components.html). This way, one component is responsible for enhancing multiple others - in our case `KibanaPageTemplate` and `NoDataConfigPage` with solution-specific code.

Reply:
Thanks, we don't use this pattern much (if at all) in EUI, so I'm just mostly unfamiliar so it took me a while to catch what was happening. Maybe just some inline comments might help?

Reply:
I find use of HOC's to be well-described in the [official docs](https://reactjs.org/docs/higher-order-components.html) and in various posts, like [this one](https://medium.com/javascript-scene/do-react-hooks-replace-higher-order-components-hocs-7ae4a08b7b58):

>You may have a poor use-case for HOCs if:
>
> - The behavior requires adding a bunch of props to a component.
> - The behavior is only used in one component.
> - The behavior must be customized for each component which uses it.
>
> You may have a good use-case for HOCs if:
> - The behavior is not specific to any single component, but rather applies to many or all components in the app, and
> - The behavior doesn’t need to provide a bunch of props to the components that use it.
> - Components can be used stand-alone without the behavior from the HOC.
> - No custom logic needs to be added to a component being wrapped by the HOC.